migration/sql-files/019-authorization-part-2.sql

```diff
@@ -0,0 +1,36 @@
+ALTER TABLE resource DROP CONSTRAINT resource_parent_resource_id_fkey;
+ALTER TABLE identity_role DROP CONSTRAINT identity_role_resource_id_fkey;
+
+ALTER TABLE resource ALTER COLUMN resource_id TYPE varchar;
+ALTER TABLE resource ALTER COLUMN parent_resource_id TYPE varchar;
+ALTER TABLE resource ALTER COLUMN name TYPE varchar;
+ALTER TABLE resource DROP COLUMN owner_id;
+
+ALTER TABLE resource ADD CONSTRAINT resource_parent_resource_id_fkey FOREIGN KEY (parent_resource_id) REFERENCES resource (resource_id);
+
+ALTER TABLE identity_role ALTER COLUMN resource_id TYPE varchar;
+ALTER TABLE identity_role ADD CONSTRAINT identity_role_resource_id_fkey FOREIGN KEY (resource_id) REFERENCES resource (resource_id);
+
+ALTER TABLE identities ADD COLUMN identity_resource_id varchar;
+ALTER TABLE identities ADD CONSTRAINT identities_identity_resource_id_fkey FOREIGN KEY (identity_resource_id) REFERENCES resource (resource_id);
+
+ALTER TABLE role ALTER COLUMN name TYPE varchar;
+
+CREATE TABLE membership (
+  member_of uuid NOT NULL references identities (id),
+  member_id uuid NOT NULL references identities (id),
```

Comment:
Is this the auto-generated ID for this table? In that case, why didn't we name it `id` instead of `member_id` ( just like `id` in the user table. :)  
No strong opinions, just curious.

Reply:
This is the identity_id I think, right?

Reply:
There is no auto-generated ID for the membership table, it is simply a many-to-many relationship between Identities.  The member_id is the identifier of the member identity, and the member_of is the identifier of the group/team/organization that it is a member of.